packages/quantic/force-app/main/default/lwc/quanticSort/quanticSort.js

```diff
@@ -148,6 +236,37 @@ export default class QuanticSort extends LightningElement {
   }
 
   get value() {
-    return this.state?.sortCriteria;
+    return this.hasCustomSortOptions
+      ? this.customSortOptions[0].value
+      : this.state?.sortCriteria;
+  }
+
+  get hasCustomSortOptions() {
+    return this.customSortOptions.length > 0;
+  }
+
+  get hasError() {
+    return this.hasInitializationError || !!this.errorMessage;
+  }
+
+  /**
+   * Returns an array of custom sort options passed via slots.
+   * @returns {SortOption[]} The specified custom sort options.
+   */
+  get customSortOptions() {
+    /** @type {SortOption[]} */
+    // @ts-ignore
+    const elements = Array.from(this.querySelectorAll('c-quantic-sort-option'));
+    if (elements.length === 0) {
+      return [];
+    }
+
+    return elements.map((option) => {
+      return {
+        label: option.label,
+        value: option.value,
+        criterion: option.criterion,
+      };
+    });
```

Comment:
Why do you need to map it btw?

```suggestion
    return elements.map(({label, value, criterion}) => ({
      label,
      value,
      criterion
    }));
```

That works too, but I don't see why you need to do this.

Do you want to ensure the returned object contains the keys `label,value,criterion` ?

Reply:
For example:

```
return Array.from(this.querySelectorAll('c-quantic-sort-option')).map(({ label, value, criterion }) => ({ label, value, criterion }));
```

This code will ensure that the returned value is an array, and explicitely will have `value: undefined` if ever one of the keys are undefined. Is that what you wanted?

Reply:
yeah that works, im basically getting the options from the slots to pass it to the combobox.